ten-simple-rules-dockerfiles.Rmd

```diff
@@ -120,10 +120,12 @@ A `Dockerfile` consists of a sequence of instructions to copy files and install
 Each instruction adds a layer to the image, which can be cached across image builds for minimizing build and download times.
 The images have a main executable exposed as an "entrypoint" that is started when they are run as stateful containers, which are the running instances of Docker images.
 Containers can be modified, stopped, restarted and purged.
-A visual analogy for building and running a container is provided in Figure&nbsp;1. Akin to compiling source code for a programming language, creating a container also starts with a plain text file (Dockerfile), which provides instructions for building an image. Similarly to using a compiled binary file to launch a program, the image is then run to create a container instance.
+A visual analogy for building and running a container is provided in Figure&nbsp;1.
+Akin to compiling source code for a programming language, creating a container also starts with a plain text file (Dockerfile), which provides instructions for building an image.
+Similarly to using a compiled binary file to launch a program, the image is then run to create a container instance.
 See Listing&nbsp;1 for a full `Dockerfile`, which we will refer to throughout this article.
 
-```{r container-analogy, echo=FALSE, fig.cap="The workflow to create Docker containers by analogy. Containers begin with a Dockerfile, a recipe for building the computational environment (analogous to source code in a compiled programming language). This is used to build an image with the `docker build` command (analogous to compiling the source code into a binary). Finally the image is used to launch one or more containers with the `docker run` command (analogous to running an instance of the compiled binary as a process).", out.width = '100%'}
+```{r container-analogy, echo=FALSE, fig.cap="The workflow to create Docker containers by analogy. Containers begin with a Dockerfile, a recipe for building the computational environment (analogous to source code in a compiled programming language). This is used to build an image with the `docker build` command (analogous to compiling the source code into an executable binary). Finally the image is used to launch one or more containers with the `docker run` command (analogous to running an instance of the compiled binary as a process).", out.width = '100%'}
```

Comment:
What is different here (I don't see it).

Reply:
Changed "compiling the source code into a binary" to "compiling the source code into an executable binary", because I'm not sure everyone immediately knows what a "binary" is.

Maybe "executable" is even a better term: https://en.wikipedia.org/wiki/Executable

@bdevans Would you be fine with saying "analogous to compiling the source code into an executable file" ?

Reply:
@nuest That would be fine I think. I chose "binary" because I wanted to convey the idea that an image is no longer plain text (like the Dockerfile that generated it) and that it can be run like an executable. 

Maybe "analogous to compiling the source code into an executable (binary) file" would be even better?

Reply:
Yes, let's go with both.